upgrade.php

```diff
@@ -392,12 +399,20 @@ function url_get_contents ($Url) {
 echo "--------------------------------------------------------\n";
 echo "STEP 4: Backing up database: \n";
 echo "--------------------------------------------------------\n\n";
-$backup = exec('php artisan snipeit:backup', $backup_results, $return_code);
-echo '-- ' . implode("\n", $backup_results) . "\n\n";
-if ($return_code > 0) {
-    die("Something went wrong with your backup. Aborting!\n\n");
+
+if (!$skip_backup) {
+    
+    try {
+        $return_code = backup();
+    } catch (Exception $e) {
+        echo 'Caught exception: '. $e->getMessage(). "\n";
+        unset($return_code);
+    }
```

Comment:
I think there should be an exit() here, so we don't proceed with the upgrade if the backup fails.  This looks like it just catches and prints the exception that backup() throws, but doesn't actually abort.